geonode/geoserver/manager.py

```diff
@@ -0,0 +1,137 @@
+        return type.objects.none()
+
+    def exists(self, uuid: str, /, instance: ResourceBase = None) -> bool:
+        if uuid and instance:
+            if hasattr(instance.get_real_instance(), 'storeType') and instance.get_real_instance().storeType not in ['tileStore', 'remoteStore']:
+                try:
+                    logger.debug(f"Searching GeoServer for layer '{instance.alternate}'")
+                    if gs_catalog.get_layer(instance.alternate):
+                        return True
+                except Exception as e:
+                    logger.debug(e)
+                    return False
+            return True
+        return False
+
+    def delete(self, uuid: str, /, instance: ResourceBase = None) -> int:
+        """Removes the layer from GeoServer
+        """
+        # cascading_delete should only be called if
+        # ogc_server_settings.BACKEND_WRITE_ENABLED == True
+        if instance and getattr(ogc_server_settings, "BACKEND_WRITE_ENABLED", True):
+            _real_instance = instance.get_real_instance()
+            if hasattr(_real_instance, 'alternate') and _real_instance.alternate:
+                if _real_instance.remote_service is None or _real_instance.remote_service.method == CASCADED:
+                    geoserver_cascading_delete.apply_async((_real_instance.alternate,))
+
+    def create(self, uuid: str, /, type: object = None, defaults: dict = {}) -> ResourceBase:
+        if type:
+            _resource = type.objects.get(uuid=uuid)
+            if type == Layer:
+                geoserver_post_save_local(_resource)
+            return _resource
+        return None
+
+    def update(self, uuid: str, /, instance: ResourceBase = None, xml_file: str = None, metadata_uploaded: bool = False,
+               vals: dict = {}, regions: dict = {}, keywords: dict = {}, custom: dict = {}, notify: bool = True) -> ResourceBase:
+        if instance:
+            if type(instance.get_real_instance()) == Layer:
+                geoserver_post_save_local(instance.get_real_instance())
+        return instance
+
+    def set_permissions(self, uuid: str, /, instance: ResourceBase = None, permissions: dict = {}, created: bool = False) -> bool:
+        # TODO: move GeoFence set perms logic here
+        return True
+
+    def set_thumbnail(self, uuid: str, /, instance: ResourceBase = None, overwrite: bool = True, check_bbox: bool = True) -> bool:
+        if instance:
+            # TODO: missing thumb for documents
+            geoserver_create_thumbnail.apply_async(((instance.id, overwrite, check_bbox, )))
+            return True
+        return False
+
+    def exec(self, method: str, uuid: str, /, instance: ResourceBase = None, **kwargs) -> ResourceBase:
+        raise NotImplementedError
+
+    def set_style(self, method: str, uuid: str, /, instance: ResourceBase = None, **kwargs) -> ResourceBase:
+        try:
+            instance = instance or Layer.objects.get(uuid=uuid)
```

Comment:
This try, since is used in two places (and can be useful in other places of the class), could be worth being extracted as a private method to reduce code redundancy.